pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/JsonConverterTests.java

```diff
@@ -114,53 +97,69 @@ public void testLogicalTypesToJson() {
                 .name("timemicros").type(timeMicrosType).noDefault()
                 .name("myuuid").type(uuidType).noDefault()
-                .name("myvarint").type(varintType).noDefault()
-                .name("myduration").type(durationType).noDefault()
                 .endRecord();
 
         final long MILLIS_PER_DAY = 24 * 60 * 60 * 1000;
 
         UUID myUuid = UUID.randomUUID();
         Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone("Europe/Copenhagen"));
 
-        GenericRecord myDecimal = new GenericData.Record(decimalType);
-        myDecimal.put("bigint", BigInteger.valueOf(123).toByteArray());
-        myDecimal.put("scale", 2);
-
-        GenericRecord myduration = new GenericData.Record(durationType);
-        myduration.put("months", 5);
-        myduration.put("days", 2);
-        myduration.put("nanoseconds", 1000 * 1000 * 1000 * 30L); // 30 seconds
-
         GenericRecord genericRecord = new GenericData.Record(schema);
-        genericRecord.put("amount", myDecimal);
         genericRecord.put("mydate", (int)calendar.toInstant().getEpochSecond());
         genericRecord.put("tsmillis", calendar.getTimeInMillis());
         genericRecord.put("tsmicros", calendar.getTimeInMillis() * 1000);
         genericRecord.put("timemillis", (int)(calendar.getTimeInMillis() % MILLIS_PER_DAY));
         genericRecord.put("timemicros", (calendar.getTimeInMillis() %MILLIS_PER_DAY) * 1000);
         genericRecord.put("myuuid", myUuid.toString());
-        genericRecord.put("myvarint", BigInteger.valueOf(12234).toByteArray());
-        genericRecord.put("myduration", myduration);
 
-        JsonNode jsonNode = JsonConverter.toJson(genericRecord);
-        assertEquals(jsonNode.get("amount").asText(), "1.23");
-        assertEquals(jsonNode.get("myvarint").asLong(), 12234);
+        JsonNode jsonNode = JsonConverter.toJson(ElasticSearchConfig.load(ImmutableMap.of()), genericRecord);
         assertEquals(jsonNode.get("mydate").asInt(), calendar.toInstant().getEpochSecond());
         assertEquals(jsonNode.get("tsmillis").asInt(), (int)calendar.getTimeInMillis());
         assertEquals(jsonNode.get("tsmicros").asLong(), calendar.getTimeInMillis() * 1000);
         assertEquals(jsonNode.get("timemillis").asInt(), (int)(calendar.getTimeInMillis() % MILLIS_PER_DAY));
         assertEquals(jsonNode.get("timemicros").asLong(), (calendar.getTimeInMillis() %MILLIS_PER_DAY) * 1000);
-        assertEquals(jsonNode.get("myduration").get("months").asInt(), 5);
-        assertEquals(jsonNode.get("myduration").get("days").asInt(), 2);
-        assertEquals(jsonNode.get("myduration").get("nanoseconds").asLong(), 30000000000L);
         assertEquals(UUID.fromString(jsonNode.get("myuuid").asText()), myUuid);
     }
+
+    @Test
+    public void testUnsupportedLogicalTypeFails() throws IOException
+    {
+        ElasticSearchConfig elasticSearchConfig = ElasticSearchConfig.load(ImmutableMap.of("ignoreUnsupportedFields", false));
```

Comment:
can we also test ignoreUnsupportedFields=true ?

Reply:
Done in JsonConverterTests.testUnsupportedLogicalTypeIgnore()